src/lib.rs

```diff
@@ -288,16 +293,24 @@ pub mod pallet {
                 .try_into()
                 .map_err(|_| Error::<T>::NameTooLong)?;
 
+            let mut bounded_participants = BoundedVec::<
+                <T as frame_system::Config>::AccountId,
+                <T as pallet::Config>::MaxPoolParticipants,
+            >::default();
+
+            ensure!(
```

Comment:
This is not needed, the owner is the first participant in the pool. If you want to check something - just check that the MaxPoolParticipants is never `0`'

Reply:
But it does not hurt, so up to you actually.

Reply:
In theory it should already know the type fro, where it puts participants into the pool. Have you tried `BoundedVec::default()`?

Reply:
We tried to use the BoundedVec::default() instead of the one specifying the types but it gives us some error when running the node. So I think we should leave it with the types.

The ensure! there was because if I just use the try_push() then it leaves an unused type Result, so i just figured could be good to put the ensure!  there just in case there is an error there, but could be change to other approach to handle the Result if you have other way to do it.

Reply:
We tried to use BoundedVec::default() but there is an issue when executing the node, but we tried with your other suggestion from below and that abreviation did work BoundedVec::<T::AccountId, T::MaxPoolParticipants>::default()